### PR TITLE
Enhancement Guided Tour: Completed tour step

### PR DESCRIPTION
**Pull request by lovaiible**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] ~~Client: I updated multiple integration tests (Jest) related to the features~~
- [x] ~~Client: I added `authorities` to all new routes and check the course groups for displaying navigation elements (links, buttons)~~
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context
This PR should implement this feature https://github.com/ls1intum/Artemis/issues/1056

### Description
After completing all steps of a tour, the completed tour step will be automatically displayed. The user can close or restart the tutorial from the last tour step.

### Steps for Testing
1. Log in to Artemis
2. Navigate to Course Overview or any other page with a tutorial
3. Start and complete tutorial

### Screenshots
image

The previous completed tour step looked like this, but @krusche did not like the gif, so I just removed it. If anyone has a suggestion for another image or gif, feel free to comment 👍 
image

### CODE REVIEW

File: src/main/webapp/app/guided-tour/guided-tour.service.ts [hunk 1]
```diff
@@ -179,6 +179,26 @@ export class GuidedTourService {
...
+
+    /**
+     * Determines if the completed tour is currently displayed
+     */
+    public isCompleteTourStep(): boolean {
+        if (this.currentTour && this.currentTour.steps) {
+            return this.currentTour.settingsKey === completedTour.settingsKey;
+        }
+        return false;
+    }
```
Comment (ungaralex) [inline src/main/webapp/app/guided-tour/guided-tour.service.ts]: You could merge those into one method having the settings key as a parameter

File: src/main/webapp/app/guided-tour/tours/general-tour.ts [hunk 2]
```diff
@@ -20,3 +20,17 @@ export const cancelTour: GuidedTour = {
...
+export const completedTour: GuidedTour = {
+    courseShortName: '',
+    exerciseShortName: '',
+    settingsKey: 'completed_tour',
+    steps: [
+        new TextTourStep({
+            headlineTranslateKey: 'tour.completed.headline',
+            contentTranslateKey: 'tour.completed.content',
+            hintTranslateKey: 'tour.completed.hint',
+            // imageUrl: 'https://media.giphy.com/media/g9582DNuQppxC/giphy.gif',
```
Comment (ungaralex) [inline src/main/webapp/app/guided-tour/tours/general-tour.ts]:
```suggestion
            // imageUrl: 'https://media.giphy.com/media/wACPZdg6nX52o/giphy.gif',
```

File: src/main/webapp/app/guided-tour/guided-tour.service.ts (hunk 1 above)
Comment (lovaiible) [reply src/main/webapp/app/guided-tour/guided-tour.service.ts]: @ungaralex I had the implementation just like suggested but then I thought that having the method names `isCancelTourStep()` and `isCompleteTourStep()` is easier to read :)

**Comment by krusche** [comment]
please resolve the merge conflicts and make sure to follow the new angular design guidelines described in #1081

File: src/main/webapp/i18n/en/guidedTourCourseOverview.json [hunk 3]
```diff
@@ -48,8 +48,8 @@
             },
             "contact": {
                 "headline": "Contact the Artemis team",
-                "content": "Feel free to contact us to leave us some feedback, request features or to report bugs in the application. If you're motivated and want to contribute to Artemis, then head over to our Github and submit your first Pull Requests",
-                "link": "Visit the Artemis Github page"
+                "content": "Feel free to contact us to leave us some feedback, request features or to report bugs in the application.",
+                "hint": "If want to contribute to Artemis, then head over to our <a href='https://github.com/ls1intum/ArTEMiS' target='_blank'>Github page</a> and submit your first Pull Requests"
```
Comment (maxr96) [inline src/main/webapp/i18n/en/guidedTourCourseOverview.json]:
```suggestion
                "hint": "If want to contribute to Artemis, then head over to our <a href='https://github.com/ls1intum/ArTEMiS' target='_blank'>Github page</a> and submit your first Pull Request"
```

File: src/main/webapp/i18n/en/guidedTourCourseOverview.json [hunk 4]
```diff
@@ -48,8 +48,8 @@
             },
             "contact": {
                 "headline": "Contact the Artemis team",
-                "content": "Feel free to contact us to leave us some feedback, request features or to report bugs in the application. If you're motivated and want to contribute to Artemis, then head over to our Github and submit your first Pull Requests",
-                "link": "Visit the Artemis Github page"
+                "content": "Feel free to contact us to leave us some feedback, request features or to report bugs in the application.",
```
Comment (maxr96) [inline src/main/webapp/i18n/en/guidedTourCourseOverview.json]:
```suggestion
                "content": "Feel free to contact us, to leave us some feedback, request features or to report bugs in the application.",
```

File: src/main/webapp/i18n/en/guidedTour.json [hunk 5]
```diff
@@ -15,6 +15,11 @@
             "headline": "Tutorial cancelled",
             "content": "You have cancelled the tutorial. If you want to continue this tutorial at any other time, then click on <tt>Continue tutorial</tt> from the account menu."
         },
+        "completed": {
+            "headline": "Congratulations 🎉",
+            "content": "You have successfully completed the tutorial. You can restart the tutorial with <tt>Restart tutorial</tt> or at any other time from the account menu dropdown",
```
Comment (maxr96) [inline src/main/webapp/i18n/en/guidedTour.json]:
```suggestion
            "content": "You have successfully completed the tutorial. You can restart the tutorial with <tt>Restart tutorial</tt> at any other time from the account menu dropdown",
```

File: src/main/webapp/app/guided-tour/guided-tour.component.scss [hunk 6]
```diff
@@ -9,12 +9,28 @@ $tour-next-button-color: #17a2b8 !default;
...
 $tour-back-button-hover: #117a8b !default;
+$tour-restart-button-color: #28a745 !default;
+$tour-restart-button-hover: #1e7e34 !default;
 $tour-button-color-disabled: #f2f2f2 !default;
 $tour-button-text-color: #ffffff !default;
 $tour-button-text-disabled: #a3a3a3 !default;
 $tour-shadow-color: #4c4c4c !default;
 
 // Mixins
+@mixin tour-button($background-color, $hover-background-color) {
```
Comment (maxr96) [inline src/main/webapp/app/guided-tour/guided-tour.component.scss]: :+1:

File: src/main/webapp/i18n/en/guidedTourCourseOverview.json (hunk 4 above)
Comment (lovaiible) [reply src/main/webapp/i18n/en/guidedTourCourseOverview.json]: @maxr96 The intention of the sentence is that the user can contact the Artemis team in order to leave some feedback, so the comma is unnecessary imo.

File: src/main/webapp/i18n/en/guidedTourCourseOverview.json (hunk 3 above)
Comment (lovaiible) [reply src/main/webapp/i18n/en/guidedTourCourseOverview.json]: @maxr96  Thanks for the suggestion. I will change the text

File: src/main/webapp/i18n/en/guidedTour.json (hunk 5 above)
Comment (lovaiible) [reply src/main/webapp/i18n/en/guidedTour.json]: @maxr96  The sentence may be misunderstanding. I will change.

**Pull request merged**